test/unit/test_console_tool.py

```diff
@@ -1891,11 +1920,54 @@ def test_sync_exclude_if_modified_after_exact(self):
             '''
 
             command = [
-                'sync', '--threads', '1', '--noProgress', '--excludeIfModifiedAfter',
-                '1367600664.152', temp_dir, 'b2://my-bucket'
+                'sync', '--noProgress', '--excludeIfModifiedAfter', '1367600664.152', temp_dir,
+                'b2://my-bucket'
             ]
             self._run_command(command, expected_stdout, '', 0)
 
+    def _test_sync_threads(
+        self, threads=None, sync_threads=None, download_threads=None, upload_threads=None
+    ):
+        self._authorize_account()
+        self._create_my_bucket()
+
+        with TempDir() as temp_dir:
+            local_file = self._make_local_file(temp_dir, 'file.txt')
+            command = [f'sync', '--noProgress']
+            if threads is not None:
+                command += ['--threads', str(threads)]
+            if sync_threads is not None:
+                command += ['--syncThreads', str(sync_threads)]
+            if download_threads is not None:
+                command += ['--downloadThreads', str(download_threads)]
+            if upload_threads is not None:
+                command += ['--uploadThreads', str(upload_threads)]
+            command += [temp_dir, 'b2://my-bucket']
+            expected_stdout = '''
+            upload file.txt
+            '''
+            self._run_command(command, expected_stdout)
+
+    def test_sync_threads(self):
+        self._test_sync_threads(threads=1)
+
+    def test_sync_sync_threads(self):
+        self._test_sync_threads(sync_threads=1)
+
+    def test_sync_download_threads(self):
+        self._test_sync_threads(download_threads=1)
+
+    def test_sync_upload_threads(self):
+        self._test_sync_threads(upload_threads=1)
+
+    def test_sync_many_thread_options(self):
+        self._test_sync_threads(sync_threads=1, download_threads=1, upload_threads=1)
+
+    def test_sync_all_thread_options(self):
+        # Using --threads is exclusive with other options
+        with self.assertRaises(ValueError):
+            self._test_sync_threads(threads=1, sync_threads=1, download_threads=1, upload_threads=1)
+
```

Comment:
That test will pass if
```
if args.syncThreads != 10 or args.uploadThreads != 10 or args.downloadThreads != 10:
```
will be mistakenly implemented as
```
if args.syncThreads != 10 and args.uploadThreads != 10 and args.downloadThreads != 10:
```
so

```suggestion
    def test_sync_thread_exclusive_options(self):
        # Using --threads is exclusive with other options
        with self.assertRaises(ValueError):
            self._test_sync_threads(threads=1, upload_threads=1)
        with self.assertRaises(ValueError):
            self._test_sync_threads(threads=1, sync_threads=1)
        with self.assertRaises(ValueError):
            self._test_sync_threads(threads=1, download_threads=1)
        with self.assertRaises(ValueError):
            self._test_sync_threads(threads=1, sync_threads=1, download_threads=1, upload_threads=1)

```